src/paho/mqtt/client.py

```diff
@@ -23,10 +23,24 @@
 import select
 import socket
 
+ssl = None
 try:
     import ssl
 except ImportError:
-    ssl = None
+    pass
+
+socks = None
+try:
+    import socks
+
+    # Python 3
+    from urllib import request as urllib_dot_request
+    from urllib import parse as urllib_dot_parse
+except ImportError:
```

Comment:
This ImportError catches 2 different kinds: one is for socks and the other is for urllib. I'd make them 2 separate try statements. I think it's OK to do the urllib imports even if socks is None. Alternatively, put the second try in the else clause of the first.